homeassistant/components/modbus/const.py

```diff
@@ -98,6 +99,16 @@ class DataType(str, Enum):
     FLOAT64 = "float64"
 
 
+class RegisterBytes(int, Enum):
+    """Register size used by sensor."""
+
+    NOT_SET = 0
+    ONE = 1
```

Comment:
Please do not allow 1 nor NOT_SET.

1 cannot be, just like you do not allow 3 or 5.

Reply:
What should happen for string and uint8/int8 ?

Reply:
"datatype: uint8/int8" is NOT allowed and have nothing to do with "datatype: string", I think you mix 2 different configurations.